flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/SnapshotManager.java

```diff
@@ -90,6 +90,22 @@ public boolean snapshotExists(long snapshotId) {
         }
     }
 
+    public @Nullable Long earlierThanTimeMills(long timestampMills) {
+        Long earliest = earliestSnapshotId();
+        Long latest = latestSnapshotId();
+        if (earliest == null || latest == null) {
+            return null;
+        }
+
+        for (long i = latest; i >= earliest; i--) {
+            long commitTime = snapshot(i).timeMillis();
```

Comment:
Is the snapshot id sure to be continuous? Otherwise we need to check if the snapshot exists

Reply:
Should be continuous, this is by design.